classes/user_manager.php

```diff
@@ -221,11 +223,12 @@ public function upsert_user(array $data) {
                     }
                 }
             } catch (\Exception $exception) {
-                throw new upset_failed_exception(
+                $exception = new upset_failed_exception(
                     'error:creating',
                     (object)['error' => $exception->getMessage()],
                     $exception->getMessage()
                 );
+                $DB->rollback_delegated_transaction($transaction, $exception);
```

Comment:
`$transaction->rollback($e);` and `$transaction->allow_commit();` look a bit nicier to me. What do you think

Reply:
what is the difference?

Reply:
ah it's pretty much the same. Yeah. Let's use your way